### ISSUE TITLE
Only require user-created structure.xml files if default child order needs to be overridden

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/islandora_compound_batch#36]
One of this module's UX characteristics is that it requires the user to generate `structure.xml` files. Is there any reason why we can't make `islandora_compound_batch_preprocess` do that for the user? The user would only need to create `structure.xml` files where the default sort order for children needed to be overridden.

**Comment by mjordan** [comment]
Tagging @bondjimbond since he and I are having this discussion via email.

**Comment by mjordan** [comment]
#24 may be relevant here.

**Comment by bondjimbond** [comment]
Alternatively: islandora_compound_batch_preprocess could (should!) (1) check for structure.xml first, and (2) if structure.xml does not exist, automatically execute create_structure_files.php on the already-set directory path as part of the preprocess.

This way, if structure.xml already exists because the user created a custom one, all proceeds happily on. If it doesn't, create it automatically, by calling the script that the user would need to run anyway.

**Comment by mjordan** [comment]
Yes, if a structure file exists, it should not be overwritten. The user would put one in the object directory it was needed, presumably.
